src/event.rs

```diff
@@ -886,7 +886,22 @@ pub enum Ime {
     /// Notifies when text should be inserted into the editor widget.
     ///
     /// Right before this event winit will send empty [`Self::Preedit`] event.
-    Commit(String),
+    Commit {
+        content: String,
+        /// If selection is Some, the selection / cursor position should be updated after
+        /// placing the `content`.
+        selection: Option<(usize, usize)>,
+        /// If compose_region is Some, the compose region should be updated after replacing the
+        /// text.
+        compose_region: Option<(usize, usize)>,
```

Comment:
Android shows a underline under the word thats currently being edited. That is the compose region. But since no other platform has this (even iOS doesn't seem to have something like this) I think we could also remove it.

Reply:
Others usually have a preedit and the update is like pending to be inserted.

Reply:
I think Android's compose region is just different terminology for the "preedit" string.

The preedit string is the transient text that would be shown that may preview a tentative completion e.g. for kanji.

I think there are essentially three high level spans of text:
1. the "surrounding" text
2. the "preedit" text (or "compose region" for Android)
3. the "selection" or "cursor" (empty selection)

I think the main thing that's changing here is that we need to introduce the notion that there is surrounding text which Winit didn't previously expose - though this isn't something unique to Android either.

Android has a slightly more general decoupling of the selection/cursor and the preedit string - while other IMEs might only track a cursor (can be considered an empty selection) or the cursor/selection will always be a subset of the preedit string.

Reply:
In the context of the Winit public API we should probably rename "compose_region" here to "preedit_region" or "preedit_span" perhaps.

Conceptually this would be committing some surrounding text and also updating the span for the selection and the span for the preedit string in one event.